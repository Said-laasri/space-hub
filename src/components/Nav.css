.links {
  width: 15rem;
  justify-content: space-between;
  align-items: center;
  text-decoration: none;
}

.links:nth-child(-n + 3) {
  text-decoration: none;
}
