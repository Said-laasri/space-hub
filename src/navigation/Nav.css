.links {
  justify-content: space-between;
  align-items: center;
  text-decoration: none;
}

.nav {
  text-decoration: none;
  font-size: 1rem;
  color: rgb(4, 101, 247);
  padding: 1rem;
}

.rockets:hover {
  text-decoration: underline;
  color: rgb(0, 174, 255);
}

.missions:hover {
  text-decoration: underline;
  color: rgb(0, 174, 255);
}

.profile:hover {
  text-decoration: underline;
  color: rgb(0, 174, 255);
}

.active {
  text-decoration: underline;
  color: rgb(1, 41, 116);
}