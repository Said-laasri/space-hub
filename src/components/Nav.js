import { Link } from 'react-router-dom';
import React from 'react';
import './Nav.css';

function Nav() {
  return (
    <div className="links d-flex">
      <Link to="/" className="reckets">
        Rockets
      </Link>
      <Link to="Mission" className="missions">
        Missions
      </Link>
      <Link to="Myprofile" className="profile">
        My Profile
      </Link>
    </div>
  );
}

export default Nav;
