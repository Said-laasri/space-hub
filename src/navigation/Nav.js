import { NavLink } from 'react-router-dom';
import React from 'react';
import './Nav.css';

function Nav() {
  return (
    <div className="links d-flex">
      <NavLink to="/" className="rockets nav" activeClassName="active">
        Rockets
      </NavLink>
      <NavLink to="Mission" className="missions nav">
        Missions
      </NavLink>
      <NavLink to="Myprofile" className="profile nav">
        My Profile
      </NavLink>
    </div>
  );
}

export default Nav;
