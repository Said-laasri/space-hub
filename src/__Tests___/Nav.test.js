import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configueStore';
import Nav from '../navigation/Nav';

it('renders correctly', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
