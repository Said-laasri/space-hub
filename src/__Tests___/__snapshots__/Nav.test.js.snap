// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="links d-flex"
>
  <a
    activeClassName="active"
    aria-current="page"
    className="rockets nav active"
    href="/"
    onClick={[Function]}
  >
    Rockets
  </a>
  <a
    className="missions nav"
    href="/Mission"
    onClick={[Function]}
  >
    Missions
  </a>
  <a
    className="profile nav"
    href="/Myprofile"
    onClick={[Function]}
  >
    My Profile
  </a>
</div>
`;
